### PR TITLE
Mateon 2024 sep

### PR DESCRIPTION
**Pull request by mateon1**
Major renames, script callbacks, cleanup

### CODE REVIEW

File: src/code.c [hunk 1]
```diff
@@ -49,13 +49,13 @@ struct unkStruct_80B7394
...
-extern void sub_800199C(u32, s32, u32, s32);
-extern void sub_80018D8(u32, u32, u32);
+extern struct unkStruct_80B7394 gScriptVarInfo[];
+extern void SetScriptVarArrayValue(u32, s32, u32, s32);
+extern void SetScriptVarValue(u32, u32, u32);
 extern void sub_80972F4(void);
 extern void nullsub_128(void);
 
-void sub_800135C(void)
+void ThouroughlyResetScriptVars(void)
```
Comment (AnonymousRandomPerson) [inline src/code.c:58]: Thouroughly -> Thoroughly

File: asm/text.s [hunk 2]
```diff
@@ -1095,8 +1095,8 @@ _08008624:
 _0800862C: .4byte gUnknown_80B86A4
 	thumb_func_end GetCharacter
 
-	thumb_func_start sub_8008630
-sub_8008630:
+	thumb_func_start HexdigitValue
+HexdigitValue:
```
Comment (DizzyEggg) [inline asm/text.s:1099]: HexDigitValue (D not capped)

**Review by DizzyEggg — COMMENTED** [review]
Also something to consider while naming string variables. In pokeemerald we usually follow the convention of `gText_`
Maybe it would be a good idea to have something similar done in this repo as well?

**Comment by SethBarberee** [comment]
Not opposed to it. I'm not heavily one side or another, to be honest.

File: asm/code_8010F28.s [hunk 3]
```diff
@@ -29,7 +29,7 @@ sub_8010F28:
 	mov r8, r1
 	ldr r4, _0801108C
 	adds r0, r4
-	bl sub_80054BC
+	bl RunAxAnimationFrame_80054BC
```
Comment (SethBarberee) [inline asm/code_8010F28.s:32]: Any reason we are keeping the address in the names? If you are fairly confident on them, I say to just take the addresses out of the name but keep a comment with the address for documentation.

**Comment by mateon1** [comment]
Sorry for the delay, review comments addressed, should be ready for merge.

**Review by AnonymousRandomPerson — APPROVED** [review]

**Comment by AnonymousRandomPerson** [comment]
Not really for or against the `gText` convention either. I'll merge this PR for now, and if anyone's up for going through and renaming all the globals then that can be its own PR.

**Pull request merged**